app/indexers/hyrax/deep_indexing_service.rb

```diff
@@ -0,0 +1,102 @@
+require 'linkeddata' # we need all the linked data types, because we don't know what types a service might return.
+module Hyrax
+  class DeepIndexingService < BasicMetadataIndexer
+    # We're overiding the default indexer in order to index the RDF labels. In order
+    # for this to be called, you must specify at least one default indexer on the property.
+    # @param [Hash] solr_doc
+    # @param [String] solr_field_key
+    # @param [Hash] field_info
+    # @param [ActiveTriples::Resource, String] val
+    def append_to_solr_doc(solr_doc, solr_field_key, field_info, val)
+      return super unless object.controlled_properties.include?(solr_field_key.to_sym)
+      case val
+      when ActiveTriples::Resource
```

Comment:
These checks break if you use the 
```ruby
class Thing
  include ActiveTriples::RDFSource
```
style instead of 
```ruby
class Thing < ActiveTriples::Resource
```
Apparently the former is now recommended?